src/Ordering.API/Program.cs

```diff
@@ -1,19 +1,19 @@
 ﻿var builder = WebApplication.CreateBuilder(args);
 
 builder.AddServiceDefaults();
-builder.AddDefaultOpenApi();
 builder.AddApplicationServices();
-
 builder.Services.AddProblemDetails();
 
-var app = builder.Build();
+var apiVersioning = builder.Services.AddApiVersioning();
 
-app.UseDefaultOpenApi();
+builder.AddDefaultOpenApi(apiVersioning);
+
+var app = builder.Build();
 
 app.MapDefaultEndpoints();
 
-app.MapGroup("/api/v1/orders")
```

Comment:
Should we keep the `RequireAuthorization` call in `Program.cs`? It strikes me that it's a useful piece of information regardless of the version of the API used.

Reply:
This one I was less sure on. If I'm not mistaken, adding `RequireAuthorization` will recursively apply to all endpoints under the associated group; regardless, of the depth of the group. That seems straight forward and makes sense. I was wondering whether all versions of an API would use the same authorization. Probably? 🤷🏽‍♂️

If you don't see any other issues with having `RequireAuthorization` in `Program.cs`, I can revert it. A future scenario I considered that would fail in a not-so-obvious way was:

```c#
app.NewVersionedApi("Orders")
   .MapOrdersApiV1()        // ← no authz
   .MapOrdersApiV2()
   .RequireAuthorization(); // ← applies to V2 only
```

If the order is reversed, I _believe_ it will work:

```c#
app.NewVersionedApi("Orders") // ← empty, top-level group
   .RequireAuthorization()    // ← applies from top-level down, recursively
   .MapOrdersApiV1()          // ← v1 group under api with authz
   .MapOrdersApiV2();         // ← v2 group under api with authz
```

Reply:
Partially reverting this back. I think the way we _recommend_ this should be done is like this:

```c#
var orders = app.NewVersionedApi("Orders");

orders.MapOrdersApiV1()
      .RequireAuthorization();

// if 2.0 was added, this is what it would look like
orders.MapOrdersApiV2()
      .RequireAuthorization();
```

It's not a hard requirement, but that makes things clear and separated.

Reply:
Partially reverting this back. I think we want to _recommend_ that it be configured like so:

```c#
var orders = app.NewVersionedApi("Orders");

orders.MapOrdersApiV1()
      .RequireAuthorization();

// if 2.0 is added, it would look like this
orders.MapOrdersApiV2()
      .RequireAuthorization();
```

This is not a hard requirement, but it keeps things separate and clear.